### ISSUE TITLE
`ndis` is always 0 for `IndexHNSWFlat` 

### ISSUE THREAD

**Issue by jtibshirani** [facebookresearch/faiss#1993]
After searching with `IndexHNSWFlat`, the reported number of distance computations `hnsw_stats.ndis` can remain 0. Looking through the code, it doesn't seem like `ndis` is being updated. It wonder if it's related to https://github.com/facebookresearch/faiss/pull/862, where the distance computers were refactored?

I noticed this while testing through the [ann-benchmarks repo](https://github.com/erikbern/ann-benchmarks), which uses `hnsw_stats.ndis` to report the number of distance computations.

Thanks for the great package and for making helpful debugging info like this available!

**Comment by mdouze** [comment]
Yes right, the distances are still collected in the IndexFlat distance computer but not copied to the stats anymore. 

https://github.com/facebookresearch/faiss/blob/master/faiss/IndexFlat.cpp#L123

Marking as an enhancement.

**Comment by pzque** [comment]
So how do we get the ndis stats for IndexHNSWFlat from Python?

> Yes right, the distances are still collected in the IndexFlat distance computer but not copied to the stats anymore.
> 
> https://github.com/facebookresearch/faiss/blob/master/faiss/IndexFlat.cpp#L123
> 
> Marking as an enhancement.
